### PR TITLE
Unified About: Add device motion and haptics

### PR DESCRIPTION
**Pull request by frosty**
Refs #17405. This PR re-introduces device motion to the about screen and also adds haptic feedback on iPhones (thanks for the suggestion @annchichi!)

https://user-images.githubusercontent.com/4780/141813384-e01f090b-4d2d-4e6f-a6ea-053566aaa93d.mov

(the above video doesn't seem to be working for me in Firefox but is fine in Safari)

I finally found a way to lock the orientation of this screen on iPhone! I don't feel like it's too much of a hack, and it's limited to only this screen so it shouldn't affect anything else. As such, I've re-added device motion support to the app bubbles. Now as you rotate your device they should track the motion of the device. You'll also feel some haptic feedback on iPhone devices as they collide.

On iPad the orientation isn't locked, so the screen will auto-rotate. However, as the device is larger this doesn't feel as bad as it did on iPhones (you only had to tilt them a little before the screen would flip).

**Known Issues**

On iPhone, if you're viewing the Me screen in landscape and tap to view the About screen, it initially displays in landscape and the balls won't move in the correct direction of gravity. Once you rotate to portrait everything will be correct. I would hope that this is a quite unlikely to happen and it'll likely resolve itself as the user rotates their device anyway.

**To test:**

* Build and run on physical devices. Test both iPhone and iPad.
* Navigate to the About screen and try tilting and rotating the device. Check that the balls move in the direction you expect.
* Double check that a few other screens continue to rotate as you'd expect – I think essentially most screens should rotate except for the login flow.

## Regression Notes

1. Potential unintended areas of impact

* Rotation in other areas of the app, but I think my changes are just limited to this one specific screen due to the extra protocol I added.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I did a smoke test of rotating various areas of the app and everything seems to work as it should.

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: .bundle/config [hunk 1]
```diff
@@ -1,3 +1,5 @@
 ---
 BUNDLE_PATH: "vendor/bundle"
+BUNDLE_RETRY: "3"
+BUNDLE_JOBS: "3"
```
Comment (Gio2018) [inline .bundle/config]: are these related to this specific PR?

File: WordPress/Classes/System/WordPressAppDelegate.swift [hunk 2]
```diff
@@ -254,6 +254,17 @@ class WordPressAppDelegate: UIResponder, UIApplicationDelegate {
         return true
     }
 
+    // Note that this method only appears to be called for iPhone devices, not iPad.
+    // This allows individual view controllers to cancel rotation if they need to.
```
Comment (Gio2018) [inline WordPress/Classes/System/WordPressAppDelegate.swift:258]: nice!

**Review by Gio2018 — APPROVED** [review]
Just one question, but for the rest :shipit: !

File: .bundle/config (hunk 1 above)
Comment (frosty) [reply .bundle/config]: No, I missed that this got committed. I'm not sure why it changed exactly (guessing a different bundler version) so I'll revert it.

**Comment by annchichi** [comment]
Great work, team! 👏👏👏. 

@frosty 

**iPhone**
- 1: It feels the gap is a bit larger. Preferably, it's 16pt. Currently, it's like it's around 34pt. 
- 2: I notice there is an arrow there. Is it possible to remove the arrow so that the move from bubbles can feel more natural? 😺 

**iPad**
- 3: It feels like the bubbles stop at the red lines. But perhaps you have already noticed. 

That's all from me. I Will add if I find more things. Thanks! 


Screen Shot 2021-11-17 at 10 24 48 PM
Screen Shot 2021-11-17 at 10 30 50 PM

**Comment by frosty** [comment]
Thanks for the feedback, @annchichi! 

1. This is just the standard grouped tableview section spacing. I can adjust it if we need, but I wonder if it would be best to stick with the default?
2. Totally, good catch, this shouldn't be here. Will get it removed!
3. This is actually on purpose. Originally, the logos went all the way to the edges of the cell, but on a wider view like the iPad it'd result in a single row of icons and it didn't look great to me:

image

But I can revert to this if you'd prefer :)

**Comment by annchichi** [comment]
Hey hey @frosty, 
1. hmm I think you are right. Let's stick with the default for now. It feels like a good reference for this one. 👍 
3. Regarding this one, I feel like it's natural that the bubbles become one-line where they can roll around or stack if the iPad is at a certain angle. The current one will make me feel like if it's a bug or they are 'blocked' by an invisible wall. Let's give it a try, how does that sound? 😃 

Thanks!
